includes/Core/Admin_Bar/Admin_Bar.php

```diff
@@ -134,41 +134,55 @@ private function add_menu_button( $wp_admin_bar ) {
 	 * @return bool True if Admin bar should display, False when it's not.
 	 */
 	public function is_active() {
-		if ( ! is_user_logged_in() || ! is_admin_bar_showing() ) {
+
+		// Only active if the admin bar is showing.
+		if ( ! is_admin_bar_showing() ) {
 			return false;
 		}
 
-		// Gets post object. On front area we need to use get_queried_object to get the current post object.
-		if ( $this->is_admin_post_screen() ) {
-			$post = get_post();
-		} else {
-			$post = get_queried_object();
-		}
+		// Determine the queried object.
+		$queried_object = get_queried_object();
 
-		if ( ! $post || ! $post instanceof \WP_Post ) {
-			return false;
-		}
+		// Checks for post objects.
+		if ( $queried_object instanceof \WP_Post ) {
 
-		if ( 'publish' !== $post->post_status ) {
-			return false;
+			// Ensure the user can view post insights for this post.
+			if ( ! current_user_can( Permissions::VIEW_POST_INSIGHTS, $queried_object->ID ) ) {
+				return false;
+			}
+
+			// Only published posts show the menu.
+			if ( 'publish' !== $queried_object->post_status ) {
+				return false;
+			}
+		} else {
+
+			// Only admins can see non post admin bar data.
+			if ( ! current_user_can( Permissions::MANAGE_OPTIONS ) ) {
```

Comment:
This should better use the (technically similar, but logically more appropriate) `Permissions::VIEW_DASHBOARD` cap.

Reply:
adjusted in https://github.com/google/site-kit-wp/pull/543/commits/3794a52aea3030deb4b60a469149c47469b384d9